src/servers/src/mysql/handler.rs

```diff
@@ -286,7 +287,27 @@ impl<W: AsyncWrite + Send + Sync + Unpin> AsyncMysqlShim<W> for MysqlInstanceShi
                     }
                     .fail();
                 }
-                let plan = replace_params_with_values(&plan, param_types, params)?;
+
+                let plan = match replace_params_with_values(&plan, param_types, &params) {
+                    Ok(plan) => plan,
+                    Err(e) => {
+                        w.error(
+                            ErrorKind::ER_TRUNCATED_WRONG_VALUE,
```

Comment:
Why shadow this error? What does it mean

Reply:
It's not shadow this error. It's returning to client. It means the parameter values and types don't match in the prepare stmt.